### PR TITLE
DEV: Improved error handling. New Instagram oEmbed endpoint. New blank value for Twitter metadata.

### PR DESCRIPTION
**Pull request by jbrw**
- Expose hash of missing data elements. Can be used to construct error/warning messages if a onebox fails.

- Use new Instagram oEmbed endpoint, if an access token is provided.

- Treat “0 minutes” as the equivalent of a blank value for Twitter metadata

### CODE REVIEW

File: lib/onebox/engine/instagram_onebox.rb [hunk 1]
```diff
@@ -15,35 +15,30 @@ def clean_url
...
 
       protected
 
+      def access_token
+        options[:facebook_app_access_token]
+      end
+
       def get_oembed_url
-        oembed_url = "https://api.instagram.com/oembed/?url=#{clean_url}"
+        if access_token && access_token != ''
```
Comment (CvX) [inline lib/onebox/engine/instagram_onebox.rb:36]:
```suggestion
        if access_token != ''
```

File: lib/onebox/engine/instagram_onebox.rb [hunk 2]
```diff
@@ -15,35 +15,30 @@ def clean_url
...
+        oembed = get_oembed
+        permalink = clean_url.gsub("/#{oembed.author_name}/", "/")
 
         { link: permalink,
-          title: "@#{author_name}",
-          image: og.image,
-          description: Onebox::Helpers.truncate(og.title, 250)
+          title: "@#{oembed.author_name}",
+          image: oembed.thumbnail_url,
+          description: oembed.title,
```
Comment (CvX) [inline lib/onebox/engine/instagram_onebox.rb:24]: No truncation needed anymore?

File: lib/onebox/engine/allowlisted_generic_onebox.rb [hunk 3]
```diff
@@ -256,6 +256,15 @@ def data
             d[:data_1] = Onebox::Helpers.truncate("#{d[:price_currency].strip} #{d[:price_amount].strip}")
           end
 
+          skip_missing_tags = [:video]
+          d.each do |k, v|
+            next if skip_missing_tags.include?(k)
+            if v == nil || v == ''
+              errors[k] ||= []
+              errors[k] = 'is blank'
```
Comment (CvX) [inline lib/onebox/engine/allowlisted_generic_onebox.rb:264]: I think it was supposed to be:

```suggestion
              errors[k] ||= []
              errors[k] << 'is blank'
```

😉

**Review by CvX — CHANGES_REQUESTED** [review]
+ the commit/PR message should be updated to reflect all non error related changes 🙂

File: lib/onebox/engine/allowlisted_generic_onebox.rb (hunk 3 above)
Comment (jbrw) [reply lib/onebox/engine/allowlisted_generic_onebox.rb]: Good catch!

**Review by CvX — APPROVED** [review]

**Pull request merged**